com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/CreateOrUpdateAppStep.java

```diff
@@ -279,6 +293,19 @@ public void printStepEndMessage() {
                                                               getStepLogger()).updateApplication(client, app);
         }
 
+        @Override
+        public void handleApplicationMetadata() {
+            if(app.getV3Metadata() != null) {

```

Comment:
I think it would be slightly better if this condition is inverted and the function returns if the metadata is null